### DISCUSSION TITLE
getting error in typescript joi validation

### DISCUSSION THREAD

**Discussion by Sameerthe-Big-O** [community/community#56382] [Programming Help] [answered]
### Select Topic Area

Bug

### Body

Here is y code of joi validation:
import { Schema } from 'joi';
import { Request, Response, NextFunction, RequestHandler } from 'express';

export const validate = <t>(schema: Schema<t>): RequestHandler => {
  (req: Request, res: Response, next: NextFunction) => {
    const result = schema.validate(req.body);
    if (result.error)
      return res.status(400).send(result.error.details[0]?.message);
    else {
      next();
    }
    return;
  };
};
But I am getting error, it's saying me 
"A function whose declared type is neither 'undefined', 'void', nor 'any' must return a value.ts(2355)"

Can anybody help?

**Comment by afaqahmad127** [comment, marked as answer]
```
import { Schema } from 'joi';
import { Request, Response, NextFunction, RequestHandler } from 'express';

export const validate = (schema: Schema): RequestHandler =>(req: Request, res: Response, next: NextFunction) => {
const result = schema.validate(req.body);
if (result.error)
return res.status(400).send(result.error.details[0]?.message);
else {
next();
}
return;
};
```


I hope it will solve your error
